streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java

```diff
@@ -152,4 +153,11 @@ default void init(final StateStoreContext context, final StateStore root) {
         // If a store doesn't implement a query handler, then all queries are unknown.
         return QueryResult.forUnknownQueryType(query, this);
     }
+
+    /**
+     * This state store's position
+     */
+    default Position getPosition() {
+        throw new UnsupportedOperationException("Not implemented");
```

Comment:
Note: this was added in another PR (not sure why it was also in this one). By adding this to the StateStore interface @vpapavas was able to dramatically simplify the change-logging store layers. However, it may be possible to avoid this as part of the same follow-on revision we discussed above.